### PR TITLE
add dm-acme 0.2.0 (and dm-env 1.3)

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
  - all on the up-and-up now!
  - ~~brought in from repo, upstream PRs created~~
    - ~~https://github.com/deepmind/acme/pull/3~~
    - ~~https://github.com/deepmind/dm_env/pull/3~~
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Some notes:

- `tree` and `env` have tests
- ~~can't build `tree` (and therefore anything else) because of `bazel` on windows~~
  - ~~https://github.com/conda-forge/bazel-feedstock/issues/36~~
- forced `env` to be `noarch: python` by ignoring EOL pythons
- ~~seeing some linkage warnings on `tree`, could be my local setup~~

### CODE REVIEW

**Comment by bollwyvl** [comment]
`acme` has a ton of optional dependencies, but many of them are to nightly builds, git remotes, etc. Will see how it develops.

**Comment by bollwyvl** [comment]
needs investigating:
```
  ERROR (dm-tree,lib/python3.7/site-packages/tree/_tree.cpython-37m-x86_64-linux-gnu.so): runpaths ['$ORIGIN/../../..'] found in /home/conda/staged-recipes/build_artifacts/dm-tree_1591122096842/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.7/site-packages/tree/_tree.cpython-37m-x86_64-linux-gnu.so

```

File: recipes/dm-acme/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    url: https://raw.githubusercontent.com/deepmind/acme/master/LICENSE
+    sha256: 9895751427d33c641865a9e210545a0efce05f8ed55fc8c9b81af66977964b0c
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  # TODO remove after bazel, dm-tree is available for windows
+  # - https://github.com/conda-forge/bazel-feedstock/issues/36
+  skip: true  # [win]
+  # noarch: python
```
Comment (scopatz) [inline recipes/dm-acme/meta.yaml:23]: This seems like it can be noarch. It doesn't compile in any deps

File: recipes/dm-tree/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  imports:
+    - tree
+  commands:
+    - pip check
+    - python -m unittest tree.tree_test tree.tree_benchmark
+  requires:
+    - pip
+    - attrs
+    - absl-py
+    - numpy
```
Comment (scopatz) [inline recipes/dm-tree/meta.yaml:43]: Does it compile against numpy?

**Comment by bollwyvl** [comment]
Hm... I tried env/acme with noarch and skip#win (as there would be missing
dependency on tree) and the linter was not amused. I can try some other
combinations.

As to numpy: in tree, it is used in its test suite only for benchmarking.

**Comment by scopatz** [comment]
RE numpy: ahh ok

**Comment by bollwyvl** [comment]
Perhaps I have to do some explicit binary patching:

https://github.com/pytorch/builder/blob/master/conda/pytorch-nightly/build.sh

**Comment by bollwyvl** [comment]
Had some more looking around the source... it does some strange things with [headers[(
https://github.com/deepmind/tree/tree/master/external). I am kinda wondering if we could get away with patching out the bazel build chain, as the c++ doesn't look like it uses much off the beaten path... will put some more effort towards it in the coming days, but any more pointers would be greatly appreciated!

**Comment by scopatz** [comment]
I have never really used bazel - I wish I could give more help here

**Comment by bollwyvl** [comment]
@scopatz no worries! It seems like quite an interesting beast.

I could imagine a future where [bazel+conda](https://github.com/bazelbuild/rules_python/issues/73) made a lot of sense for maintaining an in-house family of channels... and could even someday be useful for conda-forge level work: how slick would a [distributed build cache](https://docs.bazel.build/versions/master/remote-caching.html) be? I think we'd need a lot more [reproducibility](https://github.com/conda/conda-build/issues/2140) in the core conda/conda-build toolchain for it to work for real.

As I mentioned, it seems overkill for this package, and maybe I can figure out patching it to work with stock `compiler("c")`. I'm not super under the gun to get this out: my goal here is part of a year-long effort to build a reinforcement learning "constructoricculum" which includes state-of-the-art packages like this one, benchmark data sets, (searhable, citeable) research papers, and a jupyterlab to drive it. So I'll likely be stuck having to deal with _lots_ of crazy builds, and will have to up my game in a couple dimensions. Thanks for the support, though!

**Comment by vnlitvinov** [comment]
FYI: `dm-tree` was added in #13248 and is available now: https://anaconda.org/conda-forge/dm-tree

**Comment by bollwyvl** [comment]
Thanks for the update!

**Comment by vnlitvinov** [comment]
@bollwyvl please let me know if you wish to become a co-maintaner for dm-tree (or make a PR adding yourself there), I took tests section from your PR after all 😃

**Comment by bollwyvl** [comment]
@vnlitvinov i am generally looking to _not_ take on another exotic build chain, but feel free to ping me if you run into issues over on `tree`!

Honextly, I had forgotten that `tree` was where all the complexity was, and that `env` and `acme` can be `noarch` now. Also a ton of the extras are available now, so could add, but would be happy to see these go through in their current form. Also all the LICENSE PRs landed, so this is now a really simple PR.

Anyhow, hoping for good results on CI!

**Comment by bollwyvl** [comment]
@conda-forge/staged-recipes @conda-forge/help-python (finally) ready for review!

**Comment by vnlitvinov** [comment]
> i am generally looking to _not_ take on another exotic build chain, but feel free to ping me if you run into issues over on `tree`!

Same here :) I needed `dm-tree` as a dependency for https://github.com/conda-forge/staged-recipes/pull/11160 (of which I only need `ray-core`, but for the sake of completeness I'm going to finish packaging the whole project).

**Comment by bollwyvl** [comment]
You're a hero!

**Pull request merged**